collector/src/runtime/benchmark.rs

```diff
@@ -0,0 +1,104 @@
+use benchlib::benchmark::passes_filter;
+use cargo_metadata::Message;
+use core::option::Option;
+use core::option::Option::Some;
+use core::result::Result::Ok;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+/// A binary that defines several benchmarks using the `benchmark_suite` function from `benchlib`.
+#[derive(Debug)]
+pub struct BenchmarkSuite {
+    pub binary: PathBuf,
+    pub benchmark_names: Vec<String>,
+}
+
+impl BenchmarkSuite {
+    pub fn name(&self) -> &str {
+        self.binary.file_name().unwrap().to_str().unwrap()
+    }
+}
+
+/// A collection of benchmark suites gathered from a directory.
+#[derive(Debug)]
+pub struct BenchmarkDatabase {
```

Comment:
I don't like `Database` in this name, it makes me expect rows, or columns, or lots of data. But this is just simple metadata. Let's call this `BenchmarkSuite`.